### ISSUE TITLE
@types/winston is missing winston.createLogger

### ISSUE THREAD

**Issue by tehGoti** [DefinitelyTyped/DefinitelyTyped#20418]
- [x] I tried using the `@types/winston` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @bonnici @CodeAnimal @DABH 

The function "createLogger" to create a new logger instance isn't included in the TS definitions

**Comment by JohannesLamberts** [comment]
`const logger = new winston.Logger()` worked for me instead of
`const logger = winston.createLogger()`

The [documentation on GitHub](https://www.npmjs.com/package/winston#adding-custom-transports) might be out-of-date
The type definitions and the [documentation on npm](https://www.npmjs.com/package/winston#adding-custom-transports) seem to be accurate.

**Comment by lovell** [comment]
`createLogger` was added in winston v3.0.0-rc0, see https://github.com/winstonjs/winston/issues/1096

**Comment by alem0lars** [comment]
Is there progress on type definitions upgrade for winston?

**Comment by DABH** [comment]
Thanks for pinging on this, I'll try to work on a patch this weekend if not sooner

**Comment by jeffhuys** [comment]
@DABH Awesome, looking forward to it!

**Comment by DABH** [comment]
So the 3.x series of winston (when `createLogger` appears) is actually quite a big rewrite.  In particular, winston got split into a hierarchy of smaller libraries.  So before I can write updated typings for winston 3.x, we have to add typings for the supporting packages.  The first one is for [logform](https://github.com/winstonjs/logform) - see PR #22888 .  If anyone wants to check that out and test it and/or review it and/or +/-1 it, that would help move things along.

Once typings for `logform` are merged in, I think I'll be able to merge in typings for `winston-transport`, and after that's merged, updated typings for `winston` should be unblocked.  So it will take a bit of time, but there is a clear path at least.

Meanwhile, winston 3.x is still in RC (and has known issues), the stable 2.x branch should still work for now ;)

**Comment by Gelio** [comment]
What's the progress on this? Can I help in any way?

**Comment by DABH** [comment]
Typings for `logform` got merged in.  Didn't have quite enough time over the weekend to finish the ones for `winston-transport`, but will get to it soon (near the top of my todo list).  Then the 3.x typings should be good to go.  At some point it will be extremely useful to have people test out my typings in their apps and point out any mistakes/issues.

**Comment by DABH** [comment]
Actually, just spoke with the winston team and we're going to try to put definitions within the winston repos rather than going through DT. This is preferred by DT and makes versioning/maintenance easier/faster for everyone.  Probably best to follow ^^ that thread for further updates as we get things in shape!

**Comment by nolazybits** [comment]
any news on this please?

**Comment by andyfleming** [comment]
Looks like the latest conversation is here: https://github.com/winstonjs/winston/issues/1190

**Comment by DABH** [comment]
Correct, this should be closed.  Latest news is there^.  winston@3 will self-bundle typings without using DT.  We'll open a PR to remove winston typings from DT in the not-too-distant future, once v3 is out and typings are all done.

**Comment by DABH** [comment]
v3 typings are merged into `winston@master` now -- we can probably close this issue?  Final winston v3 won't be released till end of May, so at least by then we'll open a PR to remove the typings from DT.   Glad this is finally done, definitions should be far more maintainable from now on!

**Comment by BeauBouchard** [comment]
Now that Winston `winston@master` is at `3.0.0`, There are a few things in addition to this which I was hoping to see. 

Such as `format`,`colorize` and other stuff from the latest Winston v3 is used now to format, stylize and shape the message for the log. 

I am hoping to use simular logic and patterns in my typescript applications. (but let me know if this should be considered a separate issue)

**Comment by DABH** [comment]
You should not be using the @types/winston package anymore -- just `npm install winston` (which gets v3.0), that gets you the built-in Typescript definitions now (all the typings are in winston itself, not in DefinitelyTyped).

**Comment by BeauBouchard** [comment]
closed, Thank you.

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
